ui/src/message_popup/author/question_page.jsx

```diff
@@ -0,0 +1,284 @@
+    console.log("");
+    this.returnToQuestions();
+  },
+
+  onCreate(){
+    console.log("");
+    console.log("Creating the following new question:");
+    const studentIds = this.state.students.map(student => student.id);
+    const text = this.state.questionText;
+    const {goodExamples, badExamples} = this.getExamples();
+    const id = _.maxBy(allQuestions, question => question.id).id + 1;
+    const question = {
+      studentIds,
+      id,
+      text,
+      examples: goodExamples,
+      nonExamples: badExamples,
+      indicatorId: this.state.indicator.id,
+      indicator: this.state.indicator,
+      learningObjectiveId: this.state.learningObjective.id,
+      learningObjective: this.state.learningObjective
+    };
+    console.log(question);
+    this.returnToQuestions();
+  },
+
+  openDeleteConfirmation(){
+    this.setState({deleteConfirmationOpen: true});
+  },
+
+  closeDeleteConfirmation(){
+    this.setState({deleteConfirmationOpen: false});
+  },
+
+  onDeleteQuestion(){
+    console.log("");
+    console.log("Deleting and archiving the following question:");
+    console.log(this.props.originalQuestion);
+    console.log("");
+    this.returnToQuestions();
+  },
+
+  render(){
+    return (
+      <div>
+        <AppBar 
+          title={this.props.originalQuestion !== undefined ? `Editing Question #${this.props.originalQuestion.id}` : 'New Question'}
+          iconElementLeft={<IconButton onTouchTap={this.returnToQuestions}><ArrowBackIcon /></IconButton>}
+          />
+        <div style={styles.container}>
+          <VelocityTransitionGroup enter={{animation: 'transition.fadeIn'}} runOnMount={true}>
+            <EditingComponents.QuestionText 
+              originalText={this.props.originalQuestion !== undefined ? this.props.originalQuestion.text : undefined}
+              questionText={this.state.questionText}
+              onQuestionTextChange={this.onQuestionTextChange}
+              />
+            <EditingComponents.Students 
+              students={this.state.students}
+              addStudent={this.addStudent}
+              removeStudent={this.removeStudent}
+              />
+            <EditingComponents.Examples 
+              type="Good"
+              examplesText={this.state.goodExamplesText}
+              onExamplesChange={this.onGoodExamplesChange}
+              />
+            <EditingComponents.Examples 
+              type="Bad"
+              examplesText={this.state.badExamplesText}
+              onExamplesChange={this.onBadExamplesChange}
+              />
+            <EditingComponents.LearningObjectives 
+              learningObjective={this.state.learningObjective}
+              onChangeLearningObjective={this.onChangeLearningObjective}
+              />
+            <EditingComponents.Indicators 
+              indicator={this.state.indicator}
+              onChangeIndicator={this.onChangeIndicator}
+              />
+
+            <div style={styles.finalButtonRow}>
```

Comment:
It's up to you, but there might be enough in these buttons for it to be worth splitting into its own method.